### PR TITLE
Setup CITATION.cff

### PR DESCRIPTION
**Pull request by kevinmatthes**
This PR adds a basic setup for citation metadata powered by CITATION.cff, as discussed in #210.  (Please note that any papers citing this project using these information are yet to be written. 😉)

### CODE REVIEW

File: CITATION.cff
Comment (kevinmatthes) [inline CITATION.cff:1]: This is the basic CITATION.cff, as suggested by Aeruginous and adjusted as requested.  Please tell me whether further changes are required.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -127,3 +127,13 @@ jobs:
...
+
+  cffconvert:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: docker://citationcff/cffconvert:latest
+        with:
+          args: --validate
```
Comment (kevinmatthes) [inline .github/workflows/tests.yml:139]: This is the basic validation setup.  It shall ensure that any changes to the CITATION.cff do not affect its validity.

After the mandatory repository checkout, the official validator is called on the file.  The CFF team provides a Python 3 CLI for conversion and validation purposes which is also available as this Docker image.  As the version tag is set to `latest`, you will always have the latest available version of the validator checking the configured CITATION.cff.  Thus, this logic does not need any maintenance -- except for upcoming major updates to actions/checkout.

**Review by kevinmatthes — COMMENTED** [review]
I commented the introduced changes to explain their purpose.  Please have a look at it and tell me whether further adjustments are required.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (kevinmatthes) [reply .github/workflows/tests.yml:139]: The Docker image usually succeeds within about 10 seconds which is why I added it to the main testing CI.

**Comment by kevinmatthes** [comment]
The next step is to discuss the actual implementation of the update automation for the CITATION.cff.  The update steps are rather straightforward:  replacing the old version with the new one and updating the release date to "today".

Is there any workflow I can put the two required `sed` commands or do you use `cargo-release`?

**Comment by sourcefrog** [comment]
> The next step is to discuss the actual implementation of the update automation for the CITATION.cff. The update steps are rather straightforward: replacing the old version with the new one and updating the release date to "today".
> 
> Is there any workflow I can put the two required `sed` commands or do you use `cargo-release`?

I have just been doing it manually and thinking that I should find a tool to automate it. Rather than adding sed commands let's do that now...? Maybe `cargo-release` is a good choice. It looks like we could configure it to update this file. Filed #213

**Review by sourcefrog — APPROVED** [review]
👍🏻 to merge if you feel it's ready.

**Comment by kevinmatthes** [comment]
@sourcefrog

You can go ahead and merge this PR.  I will submit a separate PR for the automatic maintenance (#213).

**Comment by kevinmatthes** [comment]
The tests are not running because the Docker image for the validation tool is not a trusted Action, yet.  How to fix this problem, @sourcefrog?

**Comment by sourcefrog** [comment]
Thanks for diagnosing it. I can add it to the allowlist.

**Comment by kevinmatthes** [comment]
I will close and reopen this PR to have the tests run once again.

**Comment by sourcefrog** [comment]
I looked at the Actions docs and I couldn't see how to allow Docker containers.

On the whole this doesn't seem like a file that is likely to be edited very often at all (other than version bumps) and so it's not really important to have it automatically validated. So, could you remove that for now and we'll just add the file?

**Comment by kevinmatthes** [comment]
Yes, no problem.  I will remove the test. :+1:

**Comment by kevinmatthes** [comment]
I just updated the `version` and `date-released` fields to refer to today's release.  You can go ahead and merge this PR, I removed the test as you requested.

**Review by sourcefrog — APPROVED** [review]

**Pull request merged**